### PR TITLE
Vault Operator initial version

### PR DESCRIPTION
**Pull request by bonifaido**

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -266,12 +341,28 @@
   revision = "ca39e5af3ece67bbcda3d0f4f56a8e24d9f2dad4"
   version = "1.1.3"
 
+[[projects]]
+  name = "github.com/juju/ratelimit"
```
Comment (baluchicken) [inline Gopkg.lock]: This maybe will cause problem with licencing

File: cmd/vault-operator/main.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+func printVersion() {
+	logrus.Infof("Go Version: %s", runtime.Version())
+	logrus.Infof("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
+	logrus.Infof("operator-sdk Version: %v", sdkVersion.Version)
+}
+
+func main() {
+	printVersion()
+	sdk.Watch("vault.banzaicloud.com/v1alpha1", "Vault", "default", 5)
```
Comment (baluchicken) [inline cmd/vault-operator/main.go]: Do you really need to trigger an event every 5 seconds?

Comment (bonifaido) [reply cmd/vault-operator/main.go]: No, actually that is not necessary, what value would be accepted here?

Comment (baluchicken) [reply cmd/vault-operator/main.go]: 0 will trigger an event only when something happens with the watched resource

File: deploy/rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+---
+
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: default-account-vault-operator
+subjects:
+- kind: ServiceAccount
+  name: default
```
Comment (baluchicken) [inline deploy/rbac.yaml]: I think this should be "vault-operator"

Comment (bonifaido) [reply deploy/rbac.yaml]: Yes! Sorry

**Pull request merged**